### PR TITLE
Add some browsers

### PR DESCRIPTION
**Pull request by mimmi20**
These browsers were added:

- 2345Browser
- CM Browser
- Ecosia
- MobileIron
- IceCat
- Kiwi Browser
- LieBaoFast
- Mobicip
- Mint Browser
- Opera Neon
- Opera Devices
- QQ Browser Mini
- Sogou Mobile Browser
- Seznam Browser
- UC Browser Mini
- Whale Browser

For Firefox and Opera Mini the iOS variants were seperated because of the different versioning. `Chrome Webview` is seperated from Chrome because a lot of third party apps are using this UA.

### CODE REVIEW

File: regexes/client/browsers.yml [hunk 1]
```diff
@@ -81,7 +86,7 @@
     default: 'Gecko'
 
 # SeaMonkey
-- regex: '(Iceape|SeaMonkey|gnuzilla)(?:/(\d+[\.\d]+))?'
+- regex: '(Iceape|IceCat|SeaMonkey|gnuzilla|Mobicip)(?:/(\d+[\.\d]+))?'
```
Comment (sgiehl) [inline regexes/client/browsers.yml]: For v4 I will no longer allow dynamic names for browser detections. Would be better to split them up and have one regex each. That will also guarantee that the name has always the same case...

**Review by sgiehl — CHANGES_REQUESTED** [review]
Left a few comments. Also there are currently a lot tests failing. Can't merge until they are fixed.

For upcoming PRs: If you have multiple changes that are affecting many existing test fixtures it would be awesome if you could split up them up into multiple PRs. That makes it much easier to look through the changes

File: Tests/fixtures/desktop.yml [hunk 2]
```diff
@@ -5120,3 +5120,200 @@
...
+    short_name: SZ
+    version: "3.0.0"
+    engine: Blink
+    engine_version: ""
+  device:
+    type: desktop
+    brand: ""
+    model: ""
+  os_family: Windows
+  browser_family: Unknown
```
[the remark this answers is not in the document]
Comment (Findus23) [reply Tests/fixtures/desktop.yml]: Indeed it is according to https://cs.wikipedia.org/wiki/Seznam.cz_prohlížeč

File: regexes/client/browsers.yml (hunk 1 above)
Comment (mimmi20) [reply regexes/client/browsers.yml]: There is actaully no test for `gnuzilla`, and I dont have a useragent for this browser. Do you have one?

File: Tests/Parser/Client/fixtures/feed_reader.yml [hunk 3]
```diff
@@ -1,205 +1,201 @@
...
     version: "2.4"
--
+- 
   user_agent: PritTorrent/1.0
-  client:
-    type: feed reader
-    name: PritTorrent
-    version: 1.0
--
+-
```
[the remark this answers is not in the document]
Comment (mimmi20) [reply Tests/Parser/Client/fixtures/feed_reader.yml]: removed

File: Tests/fixtures/desktop.yml [hunk 4]
```diff
@@ -5120,3 +5120,200 @@
...
+    short_name: OG
+    version: "1.0.2459.0"
+    engine: Blink
+    engine_version: ""
+  device:
+    type: desktop
+    brand: ""
+    model: ""
+  os_family: Windows
+  browser_family: Unknown
```
[the remark this answers is not in the document]
Comment (mimmi20) [reply Tests/fixtures/desktop.yml]: updated

File: Tests/fixtures/desktop.yml [hunk 5]
```diff
@@ -5120,3 +5120,200 @@
...
+    short_name: OH
+    version: "4.4.22.20"
+    engine: Blink
+    engine_version: ""
+  device:
+    type: desktop
+    brand: ""
+    model: ""
+  os_family: GNU/Linux
+  browser_family: Unknown
```
[the remark this answers is not in the document]
Comment (mimmi20) [reply Tests/fixtures/desktop.yml]: updated

File: Tests/fixtures/desktop.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (mimmi20) [reply Tests/fixtures/desktop.yml]: updated

File: Tests/fixtures/desktop.yml [hunk 6]
```diff
@@ -5120,3 +5120,200 @@
...
+    short_name: QW
+    version: "5.5.1"
+    engine: Blink
+    engine_version: ""
+  device:
+    type: desktop
+    brand: ""
+    model: ""
+  os_family: Windows
+  browser_family: Unknown
```
[the remark this answers is not in the document]
Comment (mimmi20) [reply Tests/fixtures/desktop.yml]: updated

File: Tests/fixtures/phablet.yml [hunk 7]
```diff
@@ -505,17 +505,17 @@
...
+    version: "3.17.4"
     engine: Blink
     engine_version: ""
   device:
     type: phablet
     brand: M1
     model: M2 Note
   os_family: Android
-  browser_family: Chrome
+  browser_family: Unknown
```
[the remark this answers is not in the document]
Comment (mimmi20) [reply Tests/fixtures/phablet.yml]: updated

File: Tests/fixtures/phablet.yml [hunk 8]
```diff
@@ -3271,36 +3271,316 @@
...
+    short_name: LF
+    version: "3.17.4"
+    engine: Blink
+    engine_version: ""
+  device:
+    type: phablet
+    brand: M1
+    model: M2 Note
+  os_family: Android
+  browser_family: Unknown
```
[the remark this answers is not in the document]
Comment (mimmi20) [reply Tests/fixtures/phablet.yml]: updated

File: Parser/Client/Browser.php [hunk 9]
```diff
@@ -208,8 +228,8 @@ class Browser extends ClientParserAbstract
         'BlackBerry Browser' => array('BB'),
         'Baidu'              => array('BD', 'BS'),
         'Amiga'              => array('AV', 'AW'),
-        'Chrome'             => array('CH', 'BA', 'BR', 'CC', 'CD', 'CM', 'CI', 'CF', 'CN', 'CR', 'CP', 'IR', 'RM', 'AO', 'TS', 'VI', 'PT', 'AS', 'TB', 'AD', 'SB', 'WP'),
-        'Firefox'            => array('FF', 'FE', 'FM', 'SX', 'FB', 'PX', 'MB', 'EI', 'WF', 'CU', 'TF', 'QM', 'FR'),
+        'Chrome'             => array('CH', 'BA', 'BR', 'CC', 'CD', 'CM', 'CI', 'CF', 'CN', 'CR', 'CP', 'IR', 'RM', 'AO', 'TS', 'VI', 'PT', 'AS', 'TB', 'AD', 'SB', 'WP', 'I3', 'CV'),
+        'Firefox'            => array('FF', 'FE', 'FM', 'SX', 'FB', 'PX', 'MB', 'EI', 'WF', 'CU', 'TF', 'QM', 'FR', 'I4'),
```
[the remark this answers is not in the document]
Comment (mimmi20) [reply Parser/Client/Browser.php]: updated

File: Tests/fixtures/phablet.yml [hunk 10]
```diff
@@ -3271,36 +3271,316 @@
...
+    short_name: LF
+    version: "3.17.4"
+    engine: Blink
+    engine_version: ""
+  device:
+    type: phablet
+    brand: XI
+    model: Redmi Note 2
+  os_family: Android
+  browser_family: Unknown
```
[the remark this answers is not in the document]
Comment (mimmi20) [reply Tests/fixtures/phablet.yml]: updated

File: regexes/client/browsers.yml (hunk 1 above)
Comment (sgiehl) [reply regexes/client/browsers.yml]: We could use this one: https://developers.whatismybrowser.com/useragents/parse/210967-linux-gecko

But actually I'm not sure if "browser" would be the correct classification for GNUZilla, as it actually was a application suite containing multiple tools like browser, email, .... Maybe we should move it to pim, or remove it completely 🤔 
Same actually applies to Iceape and SeaMonkey. All of them were modified versions of the `Mozilla Application Suite`

File: Tests/Parser/Client/fixtures/browser.yml [hunk 11]
```diff
@@ -1862,21 +1862,201 @@
...
+    engine: ""
+    engine_version: ""
+- 
+  user_agent: Mozilla/5.0 (Windows NT 6.1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/52.0.2743.116 Whale/0.7.33.5 Safari/537.36
+  client:
+    type: browser
+    name: Whale Browser
+    short_name: WH
+    version: "0.7.33.5"
+    engine: WebKit
```
[the remark this answers is not in the document]
Comment (mimmi20) [reply Tests/Parser/Client/fixtures/browser.yml]: updated

**Comment by mimmi20** [comment]
I moved SeaMoney and Iceape to the pim section, and removed gnuzilla. The mentioned useragent matched to IceCat.

File: Tests/Parser/Client/fixtures/browser.yml [hunk 12]
```diff
@@ -665,34 +665,31 @@
...
   user_agent: Mozilla/5.0 (X11; Linux x86_64; rv:10.0.12) Gecko/20130823 Firefox/10.0.11esrpre Iceape/2.7.12
   client:
-    type: browser
-    name: Iceape
-    short_name: IA
+    type: pim
+    name: SeaMonkey
     version: "2.7.12"
-    engine: Gecko
-    engine_version: ""
```
Comment (sgiehl) [inline Tests/Parser/Client/fixtures/browser.yml]: The seamonkey test cases needs to be moved to the pim fixtures

File: Parser/Client/Browser.php [hunk 13]
```diff
@@ -94,12 +99,14 @@ class Browser extends ClientParserAbstract
         'FN' => 'Fireweb Navigator',
         'GA' => 'Galeon',
         'GE' => 'Google Earth',
+        'GZ' => 'gnuzilla',
```
Comment (sgiehl) [inline Parser/Client/Browser.php]: That one can be removed.

Comment (mimmi20) [reply Parser/Client/Browser.php]: removed

File: Tests/Parser/Client/fixtures/browser.yml (hunk 12 above)
Comment (mimmi20) [reply Tests/Parser/Client/fixtures/browser.yml]: moved.

**Comment by sgiehl** [comment]
awesome. tests are passing again. will merge now

**Pull request merged**